evm-ext/src/state/block_hash.rs

```diff
@@ -6,8 +6,17 @@ pub trait BlockHashLookup {
     fn hash_by_number(&self, number: u64) -> Option<B256>;
 }
 
+/// A trait for 0x40 BLOCKHASH opcode retrieval
+pub trait BlockHashWriter {
+    fn push(&mut self, height: u64, hash: B256);
+}
+
 impl BlockHashLookup for () {
     fn hash_by_number(&self, _number: u64) -> Option<B256> {
         None
```

Comment:
Isn't this supposed to return the block hash?

Reply:
Yes if we're trying to replicate hashmap's entry interface and to have a check if the value was already present in the cache, which sounds like a good idea. But it wraps around to the concern I've outlined [here](https://github.com/UmiNetwork/op-move/pull/375#discussion_r2149718258).